pages/create/index.page.js

```diff
@@ -1,35 +1,78 @@
-import Button from "../../components/button/button";
-import Input from "../../components/input/input";
-import Select from "../../components/select/select";
-import { Container, Form, Title } from "./create.styles";
 import { useForm } from "react-hook-form";
 import { number, object, string, date, ref } from "yup";
 import { yupResolver } from "@hookform/resolvers/yup";
+import { useContractWrite, usePrepareContractWrite } from "wagmi";
+import useDebounce from "../../hooks/useDebounce";
+
+import Select from "../../components/select/select";
+import crowdfundingConfig from "../../crowdfunding.config.json";
+
+import { Container, Form, Title } from "./create.styles";
+
+const validationSchema = object().shape({
+  title: string().required(),
+  subtitle: string().required(),
+  story: string().required(),
+  token: string().required(),
+  fundGoal: number().required(),
+  startDate: date()
+    .required()
+    .min(new Date(), "Date should be greater than now"),
+  endDate: date()
+    .required()
+    .min(ref("startDate"), "End date should be after start date"),
+});
 
 const Create = () => {
-  const validationSchema = object().shape({
-    title: string().required(),
-    subtitle: string().required(),
-    story: string().required(),
-    token: string().required(),
-    fundGoal: number().required(),
-    startDate: date()
-      .required()
-      .max(new Date(), "Created date can not be future"),
-    endDate: date()
-      .required()
-      .min(ref("startDate"), "End date should be after start date"),
-  });
+  const { abi, address } = crowdfundingConfig;
 
   const formOptions = {
     resolver: yupResolver(validationSchema),
     defaultValues: {},
     mode: "onChange",
   };
-  const { handleSubmit, reset, register, formState } = useForm(formOptions);
-  const { errors } = formState;
+  const {
+    handleSubmit,
+    reset,
+    register,
+    formState: { errors },
+    getValues,
+    setValue,
+  } = useForm(formOptions);
+  const debouncedStartDate = useDebounce(
```

Comment:
out of curiosity, you're using `useDebounce` here to avoid setting the value on each `on change` event of the form right? 

Does it really worth it or is a hook that we need to maintain for a little advantage? I mean, for use cases like a search functionality that hits an Api makes sense (since you don't want to set the result value until it fetched) but here is just setting a value from a date picker and a input 

which was your rationale to take this desision ?

Reply:
@marcosmartinez7 I was following this example, here is an explanation https://wagmi.sh/examples/contract-write-dynamic#step-5-add-a-debounce-to-the-input-value

Reply:
Ohhh I see! "RPC call on every change to args", that's crazy haha